datavec/datavec-python/src/main/java/org/datavec/python/NumpyArray.java

```diff
@@ -0,0 +1,114 @@
+package org.datavec.python;
+
+import org.apache.commons.lang3.ArrayUtils;
+import org.bytedeco.javacpp.Pointer;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.nd4j.linalg.api.buffer.DataBuffer;
+import org.nd4j.linalg.api.buffer.util.DataTypeUtil;
+import org.nd4j.linalg.api.ndarray.INDArray;
+import org.bytedeco.javacpp.FloatPointer;
+import org.bytedeco.javacpp.DoublePointer;
+import org.nd4j.linalg.dataset.DataSet;
+import org.nd4j.linalg.factory.Nd4j;
+import org.nd4j.nativeblas.NativeOps;
+import org.nd4j.nativeblas.NativeOpsHolder;
+import org.nd4j.linalg.api.buffer.DataType;
+
+
+import java.util.ArrayList;
+
+
+public class NumpyArray {
+
+    private static NativeOps nativeOps = NativeOpsHolder.getInstance().getDeviceNativeOps();
+    private long address;
+    private long[] shape;
+    private long[] strides;
+    private DataType dtype = DataType.FLOAT;
+    private INDArray nd4jArray;
+
+    public NumpyArray(long address, long[] shape, long strides[], boolean copy){
+        this.address = address;
+        this.shape = shape;
+        this.strides = strides;
+        setND4JArray();
+        if (copy){
+            nd4jArray = nd4jArray.dup();
+            this.address = nd4jArray.data().address();
+
+        }
+    }
+    public NumpyArray(long address, long[] shape, long strides[]){
+        this(address, shape, strides, false);
+    }
+
+    public NumpyArray(long address, long[] shape, long strides[], DataType dtype){
+        this(address, shape, strides, dtype, false);
+    }
+
+    public NumpyArray(long address, long[] shape, long strides[], DataType dtype, boolean copy){
+        this.address = address;
+        this.shape = shape;
+        this.strides = strides;
+        this.dtype = dtype;
+        setND4JArray();
+        if (copy){
+            nd4jArray = nd4jArray.dup();
+            this.address = nd4jArray.data().address();
+        }
+    }
+
+    public long getAddress() {
```

Comment:
lombokify this maybe? We can probably get rid of the getters and setters with the annotation.